content/fundamentals/advanced.md

```diff
@@ -0,0 +1,93 @@
+### Advanced Concepts
+
+So far we've breezed over a couple of advanced topics in TypeScript like decorators and how the DI system is actually working under the hood. If all you want is a system that works, ou can probably move on to another section, but if you want to get a deeper understanding of how Nest works, this is the section for you.
+
+#### The @Injectable() Decorator
+
+Earlier in the docs we mentioned that "The `@Injectable()` decorator attaches metadata, which declares that `CatsService` is a class that can be managed by the Nest IoC container.". This is only a half truth, in reality, adding the provider to a module's `providers` array is what makes the provider able to be injected via the IoC container. What `@njectable()` is doing is forcing TypeScript to emit metadata about the class's `constructor`, specifically the `design:paramtypes` metadata that will be read at start up. If there is an `@Inject()` decorator in the constructor, by technicality this does enough to make TypeScript emit all the same metadata. Take a look at the compiled JavaScript from the following TypeScript class:
```

Comment:
Some small corrections  😄

Reply:
"This is only a half truth," Since that's a full sentence on it's own, there should be a conjunction here ("and") or that comma should be a period or semi colon.